### DISCUSSION TITLE
Keeping status checks up to date would be easier if the required status checks could be a wildcard

### DISCUSSION THREAD

**Discussion by leighmcculloch** [community/community#12377] [Actions]
Keeping status checks up to date would be easier if the required status checks could be a wildcard.

It's already possible to wildcard on branches, which is super useful for defining a small number of protection rules that apply to multiple branches.

It would be equally useful if wildcards were supported in the status checks too, so that I could say that status checks with name `test *` are required, that way all the test jobs with all their matrix values are automatically required to pass.

I work in a repo https://github.com/stellar/go that have some growing matrix setups, and the names for the checks are getting really long and have to be updated by an admin every time we add a new matrix value.

**Comment by ethomson** [comment]
Thanks @leighmcculloch - I can see how this would be useful indeed. I'll make sure that this is on the backlog.

**Reply by raygesualdo** [reply]
We looked into using dynamic matrices for our tests as well (spinning up 1-N test jobs based on certain criteria) and not being able to wildcard our required checks was a hard blocker.

**Reply by azu** [reply]
GitHub's automerge require protect branches.

- [Automatically merging a pull request - GitHub Docs](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/automatically-merging-a-pull-request)

I want to use "Require status checks to pass before merging" as trigger for automerge. (I do not want to use approve count because almost repo related to personal development.)
However, I use always matrix workflow like `Test on Node.js <version>`.
This matrix need to update regulary. As a result, It is hard to "Require status checks to pass before merging". 

<img width="817" alt="image" src="https://user-images.githubusercontent.com/19714/212442074-1cd49a9d-f746-4ed6-ab6a-61508b8faf5d.png">

wildcard could resolve it.

**workaround**

Add empty action like `name: PASS` for automerge. I think this approach is a waste of resources.

**Reply by mauritsdebruin** [reply]
We really need this too. We have dozens of project with different kinds of jobs based on project type and with matrices to test builds across different environments.

If we could wildcard all these jobs inside our status check it would makes things so much easier.. in fact it's nearly unmaintainable without a wildcard at the moment.

**Reply by Atry** [reply]
I think if you have many repositories, a GitHub Action to run Terraform to update the repository settings might help.

**Reply by espipj** [reply]
is there any timeline for this feature to be released? Thank you

**Reply by gal-rosencovich** [reply]
That indeed will be a helpful feature. Wonder if there is a timeline for this

**Reply by lee5i3** [reply]
I agree, we have some of our status checks plans from Terraform Cloud, it would be nice to be able to only have a status check if all of them pass, not having to specify each workspace

**Reply by mistercrunch** [reply]
Also need this. I like the flexibility of the support for a wildcard/regex, but I'd like to say that supporting the job name of the matrix itself seems important too. It's not intuitive that `required_status_checks` has a mix of pure job names AND "matrix instances" with a special formatting. I'd say most users would expect consistency is the input being job name and not job instances. Also I doubt people have only a subset of matrix instances that are required, in most cases the requirement is that the whole matrix passes the checks.

**Reply by seemethere** [reply]
It's actually weird that Github doesn't directly support this. You would think this would be a baseline feature for required status checks otherwise it's basically useless for anyone with a testing matrix larger than 1.

**Reply by mistercrunch** [reply]
Recently migrated a project that had many actions with a python matrix of `["3.9", "3.10"]` to ["3.10", "3.11"] and it's easy to trip all over yourself. You have to create no-op fake actions to satisfy the condition, and once it lands you have to rebase every open PR to align the jobs with the checks that are then on your main branch. To work around this we moved to calling the matrix `["current", "next"]` so it'd be easier next time, but being able to have a wildcard on the check would be good. 

Oh other request while I'm here, would be some sort of "required, but only if triggered". Use case is say I have a folder `backend/` and `frontend/` and a unit test suite for each. I want to only trigger each suite if something has changed in the corresponding folder, and I want it to be required to pass, but only if it ran. Currently there's no easy way to specify this, you have to add some sort of no-op logic in your actions or other hacks.

**Reply by gunar** [reply]
> I agree, we have some of our status checks plans from Terraform Cloud, it would be nice to be able to only have a status check if all of them pass, not having to specify each workspace

@lee5i3, did you end up implementing this? Did it work? If then, are you able to share some code? Thanks in advance.

**Comment by sebastian-fredriksson-bernholtz** [comment]
There are **a lot** of related discussions, as well as workarounds and community solutions around this. 

I can recommend https://github.com/upsidr/merge-gatekeeper (referred in https://github.com/orgs/community/discussions/26251) that allows you to specify a single workflow that will fail if any of the other checks triggered on a commit fails. It basically acts as a wildcard would, but it also allows you to explicitly specify workflows/checks to ignore (ie ones you don't want to be blocking).

It basically does what you would expect the built-in check would be doing: block if any triggered check fails. Would be nice if github added native support for it though.

**Comment by neilime** [comment]
Hi Github team,

Any chance to have this wildcard check settings in a near future?

I have many use cases where "hardcoding" is complex and worst sometimes risky.

Example:

I have a ci check based on a matrix (running checks on different language version). 
So I have to define all matrix jobs in required status checks: painful...
If there is a new version to check, we have to dont forget to add it in status checks: **RISKY**!

Thanks

**Comment by mistercrunch** [comment]
FYI - I attended GitHub's "Maintainer's Summit" today and trying to get GitHub's attention on this particular issue and a few other ones. I think I have a contact at GitHub now and hoping we can attention.

**Comment by Hau-Hau** [comment]
Any plans to support this feature? It is hard to build anything more complex without wildcards.

**Comment by das-mgrimstad** [comment]
I'm also wondering if there are any plans to support this. Even if it is not planned that would be good to know

**Comment by willemkokke** [comment]
Just ran into this too setting up a compiled python extension for python 3.10 to 3.14 for multiple platforms and architectures...

**Comment by vertti** [comment]
Really does not seem like something we should wait 3+ years for.

**Comment by silverwind** [comment]
No wildcard support is especially annoying for:

- Sharding: `test (1)`, `test (2)`
- Variations: `test-foo`, `test-bar`

With wildcards these could simply be `test*`, but without, it needs 4 rules and constant admin attention when those change.

**Comment by giomf** [comment]
This is such a basic feature that I was expecting Github supporting it.

Silly me.
